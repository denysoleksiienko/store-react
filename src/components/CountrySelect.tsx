import React from 'react';
import { Typeahead } from 'react-bootstrap-typeahead';
import { ICountrySelect } from 'interfaces/Countries';

export const CountrySelect: React.FC<ICountrySelect> = ({
  placeholder,
  onChange,
  options,
  selected,
  isValid,
  isInvalid,
}) => (
  <Typeahead
    id="basic-typeahead-single"
    placeholder={placeholder}
    onChange={onChange}
    options={options}
    selected={selected}
    isValid={isValid}
    isInvalid={isInvalid}
  />
);
