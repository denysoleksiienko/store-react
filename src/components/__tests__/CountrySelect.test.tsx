import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Typeahead } from 'react-bootstrap-typeahead';

import { CountrySelect } from '../CountrySelect';

const props = {
  id: 'basic-typeahead-single',
  placeholder: 'Select Country',
  onChange: jest.fn(),
  options: [{ label: 'Austria', value: 'AT' }],
  selected: [{ label: 'Austria', value: 'AT' }],
  isValid: true,
  isInvalid: false,
};

describe('CountrySelect component', () => {
  it('should be defined CountrySelect', () => {
    const wrapper = shallow(<CountrySelect {...props} />);
    expect(wrapper).toBeDefined();
  });

  it('should be defined Typeahead', () => {
    const wrapper = shallow(<Typeahead {...props} />);
    expect(wrapper).toBeDefined();
  });

  it('should match snapshot', () => {
    const wrapper = renderer.create(<CountrySelect {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
