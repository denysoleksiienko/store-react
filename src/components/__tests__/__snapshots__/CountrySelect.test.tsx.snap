// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelect component should match snapshot 1`] = `
<div
  className="rbt"
  style={
    Object {
      "outline": "none",
      "position": "relative",
    }
  }
  tabIndex={-1}
>
  <div
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "height": "100%",
        "position": "relative",
      }
    }
  >
    <input
      aria-autocomplete="both"
      aria-expanded={false}
      aria-haspopup="listbox"
      autoComplete="off"
      className="rbt-input-main form-control rbt-input is-valid"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Select Country"
      role="combobox"
      type="text"
      value="Austria"
    />
    <input
      aria-hidden={true}
      className="rbt-input-hint"
      readOnly={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "boxShadow": "none",
          "color": "rgba(0, 0, 0, 0.35)",
          "left": 0,
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      tabIndex={-1}
      value=""
    />
  </div>
</div>
`;
